corehq/messaging/scheduling/scheduling_partitioned/models.py

```diff
@@ -673,6 +673,12 @@ class CaseTimedScheduleInstance(CaseScheduleInstanceMixin, AbstractTimedSchedule
     # See corehq.apps.data_interfaces.models.CreateScheduleInstanceActionDefinition.reset_case_property_name
     last_reset_case_property_value = models.TextField(null=True)
 
+    def additional_deactivation_condition_reached(self):
+        from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+
+        rule = AutomaticUpdateRule.objects.get(pk=self.rule_id)
+        return rule.criteria_match(self.case_id, datetime.utcnow())
+
```

Comment:
Hey @pxwxnvermx 
1. Do we need the same change for `CaseAlertScheduleInstance`?
2. Additionally, this is overriding `additional_deactivation_condition_reached` from `CaseScheduleInstanceMixin` so that should be checked as well, I guess the order does not matter.
3. For readability, the function should be after the Meta subclass.

Reply:
Thanks for the inputs @mkangia.

1. `CaseAlertScheduleInstance` class handles the Conditional Alerts which have immediate schedules, the ticket was focused on broadcast schedules and conditional alerts being propagated when the broadcast schedules was valid but condition is False, so I added this check to the CaseTimedScheduleInstance only.
2. fixed https://github.com/dimagi/commcare-hq/pull/32671/commits/e32b76c1ca269eeb178bbc99ac4a660e9fca3955
3. fixed https://github.com/dimagi/commcare-hq/pull/32671/commits/806a1b3ca01ef9b15e17995c05fd65eb1e6db58c